File: sdk/fsharp/src/HiTechCloudSDK/SeoAndWeb.fs
namespace HiTechCloud.SDK

/// SEO & Web
type SeoAndWebResource(httpClient: HttpClient) =

    /// GET /api/domain/rank/batch - Tra hạng cho tối đa 100 tên miền trong một lượ...
    member this.DomainRankBatch() =
        async {
            return! httpClient.getAsync("/api/domain/rank/batch")
        }

    /// POST /api/domain/rank/top - Nhóm tên miền dẫn đầu
    member this.DomainRankTop() =
        async {
            return! httpClient.postAsync("/api/domain/rank/top")
        }

    /// GET /api/domain/rank - Hạng độ phổ biến của một tên miền trong bảng m...
    member this.DomainRank() =
        async {
            return! httpClient.getAsync("/api/domain/rank")
        }

    /// GET /api/tools/seo/meta/tags - Đọc thẻ meta
    member this.ToolsSeoMetaTags() =
        async {
            return! httpClient.getAsync("/api/tools/seo/meta/tags")
        }

    /// GET /api/tools/seo/robots/checker - Đọc và kiểm tra robots
    member this.ToolsSeoRobotsChecker() =
        async {
            return! httpClient.getAsync("/api/tools/seo/robots/checker")
        }

    /// GET /api/tools/seo/sitemap/extractor - Đọc sitemap
    member this.ToolsSeoSitemapExtractor() =
        async {
            return! httpClient.getAsync("/api/tools/seo/sitemap/extractor")
        }

    /// GET /api/tools/seo/tech/stack - Nhận diện công nghệ đang chạy trên website
    member this.ToolsSeoTechStack() =
        async {
            return! httpClient.getAsync("/api/tools/seo/tech/stack")
        }

    /// GET /api/tools/seo/user/agent/parser - Bóc tách chuỗi User-Agent thành trình duyệt và...
    member this.ToolsSeoUserAgentParser() =
        async {
            return! httpClient.getAsync("/api/tools/seo/user/agent/parser")
        }

    /// GET /api/tools/web/meta/tags - Đọc thẻ meta của một trang
    member this.ToolsWebMetaTags() =
        async {
            return! httpClient.getAsync("/api/tools/web/meta/tags")
        }

    /// POST /api/tools/web/open/graph - Sinh bộ thẻ Open Graph chuẩn cho một trang
    member this.ToolsWebOpenGraph() =
        async {
            return! httpClient.postAsync("/api/tools/web/open/graph")
        }

    /// POST /api/tools/web/screenshot - Chụp ảnh màn hình một trang web theo kích thướ...
    member this.ToolsWebScreenshot() =
        async {
            return! httpClient.postAsync("/api/tools/web/screenshot")
        }

    /// POST /api/tools/web/tech/stack - Nhận diện công nghệ website
    member this.ToolsWebTechStack() =
        async {
            return! httpClient.postAsync("/api/tools/web/tech/stack")
        }

File: sdk/clojure/src/hitechcloud/sdk/seo_and_web.clj
(ns hitechcloud.sdk.seo_and_web
  (:require [hitechcloud.sdk.http-client :as http]))

;; SEO & Web

(defn domain_rank_batch "GET /api/domain/rank/batch - Tra hạng cho tối đa 100 tên miền trong một lượ..." [client & [params]] (http-get client "/api/domain/rank/batch" params))

(defn domain_rank_top "POST /api/domain/rank/top - Nhóm tên miền dẫn đầu" [client & [params]] (http-post client "/api/domain/rank/top" params))

(defn domain_rank "GET /api/domain/rank - Hạng độ phổ biến của một tên miền trong bảng m..." [client & [params]] (http-get client "/api/domain/rank" params))

(defn tools_seo_meta_tags "GET /api/tools/seo/meta/tags - Đọc thẻ meta" [client & [params]] (http-get client "/api/tools/seo/meta/tags" params))

(defn tools_seo_robots_checker "GET /api/tools/seo/robots/checker - Đọc và kiểm tra robots" [client & [params]] (http-get client "/api/tools/seo/robots/checker" params))

(defn tools_seo_sitemap_extractor "GET /api/tools/seo/sitemap/extractor - Đọc sitemap" [client & [params]] (http-get client "/api/tools/seo/sitemap/extractor" params))

(defn tools_seo_tech_stack "GET /api/tools/seo/tech/stack - Nhận diện công nghệ đang chạy trên website" [client & [params]] (http-get client "/api/tools/seo/tech/stack" params))

(defn tools_seo_user_agent_parser "GET /api/tools/seo/user/agent/parser - Bóc tách chuỗi User-Agent thành trình duyệt và..." [client & [params]] (http-get client "/api/tools/seo/user/agent/parser" params))

(defn tools_web_meta_tags "GET /api/tools/web/meta/tags - Đọc thẻ meta của một trang" [client & [params]] (http-get client "/api/tools/web/meta/tags" params))

(defn tools_web_open_graph "POST /api/tools/web/open/graph - Sinh bộ thẻ Open Graph chuẩn cho một trang" [client & [params]] (http-post client "/api/tools/web/open/graph" params))

(defn tools_web_screenshot "POST /api/tools/web/screenshot - Chụp ảnh màn hình một trang web theo kích thướ..." [client & [params]] (http-post client "/api/tools/web/screenshot" params))

(defn tools_web_tech_stack "POST /api/tools/web/tech/stack - Nhận diện công nghệ website" [client & [params]] (http-post client "/api/tools/web/tech/stack" params))

